src/Ghcid.hs

```diff
@@ -192,29 +200,41 @@ mainWithTerminal termSize termOutput =
                 opts <- return $ opts{restart = nubOrd $ (origDir </> ".ghcid") : restart opts, reload = nubOrd $ reload opts}
                 when (topmost opts) terminalTopmost
 
-                termSize <- return $ case (width opts, height opts) of
-                    (Just w, Just h) -> return $ TermSize w h WrapHard
+                termSize <- case (width opts, height opts) of
+                    (Just w, Just h) -> return $ TermSize w (Just h) WrapHard
                     (w, h) -> do
                         term <- termSize
                         -- if we write to the final column of the window then it wraps automatically
                         -- so putStrLn width 'x' uses up two lines
                         return $ TermSize
                             (fromMaybe (pred $ termWidth term) w)
-                            (fromMaybe (termHeight term) h)
+                            (h <|> termHeight term)
                             (if isJust w then WrapHard else termWrap term)
 
+                termSize <- return $
+                  if no_height_limit opts
+                  then termSize { termHeight = Nothing }
+                  else termSize
+
+                supportsANSI <- hSupportsANSI stdout
+
                 restyle <- do
-                    useStyle <- case color opts of
-                        Always -> return True
-                        Never -> return False
-                        Auto -> hSupportsANSI stdout
+                    let useStyle = case color opts of
+                          Always -> True
+                          Never -> False
+                          Auto -> supportsANSI
                     when useStyle $ do
                         h <- lookupEnv "HSPEC_OPTIONS"
                         when (isNothing h) $ setEnv "HSPEC_OPTIONS" "--color" -- see #87
                     return $ if useStyle then id else map unescape
 
+                clear <- return $
```

Comment:
They wrote `--clear`. If their terminal thinks it doesn't support ANSI, I imagine they still want it to clear anyway. I'd skip the supportsANSI check here.